grammar.js

```diff
@@ -205,46 +205,7 @@ module.exports = grammar({
       seq("->", repeat1(alias($._decimal_integer, $.integer))),
 
     asm_function_body: ($) =>
-      seq(
-        "{",
-        prec.right(
-          repeat(
-            choice(
-              // list with { }
-              $.asm_list,
-              // others
-              $._asm_instruction,
-            ),
-          ),
-        ),
-        prec.right("}"),
-      ),
-
-    asm_list: ($) => seq("{", /\s/, repeat($._asm_instruction), "}", /\s/),
-
-    _asm_instruction: ($) =>
-      choice(
-        // string
-        $._asm_string,
-        // char
-        seq("char", /\s/, /\S/, /\s/),
-        // custom
-        /\S+/, // NOTE: this point can be significantly improved
-      ),
-
-    _asm_string: (_) =>
-      seq(
-        choice('abort"', '."', '+"', '"'),
-        token.immediate(prec(1, /[^"]+/)),
-        token.immediate('"'),
-        /\s/,
-      ),
-
-    // NOTE: May be re-introduced in the future, unused in the current parser
-    // listNoStateCheck
-    // seq("({)", /\s/, repeat($._asm_instruction), "(})", /\s/),
-    // hexLiteral
-    // _asm_hex_literal: (_) => /[xB]\{[\s\da-fA-F]*_?\s*\}\s/,
+      seq("{", prec.right(repeat($._tvm_statement)), prec.right("}")),
```

Comment:
We struggled a bit with what can be done with assembly and ended up with this grammar.

https://github.com/tact-lang/tact/blob/main/src/grammar/next/grammar.gg#L162-L168

The thing that worries me is that these two grammars do not match.